gradio/interface.py

```diff
@@ -476,7 +477,80 @@ def __repr__(self):
         return repr
 
     def get_config_file(self):
-        return utils.get_config_file(self)
+        components = []
+        layout = {"id": 0, "children": []}
+        dependencies = []
+
+        def add_component(parent, component):
+            id = len(components) + 1
+            components.append(
+                {
+                    "id": len(components) + 1,
+                    "type": component.__class__.__name__.lower(),
+                    "props": component.get_template_context(),
+                }
+            )
+            layout_context = {"id": id}
+            if isinstance(component, BlockContext):
+                layout_context["children"] = []
+            parent["children"].append(layout_context)
+            return layout_context
+
+        if self.title:
+            add_component(layout, Markdown("<h1>" + self.title + "</h1>"))
+        if self.description:
+            add_component(layout, Markdown(self.description))
+        panel_row = add_component(layout, Row())
+        input_panel = add_component(panel_row, Column())
+        input_ids = []
+        for component in self.input_components:
+            input_id = add_component(input_panel, component)
```

Comment:
better have this for clarity
input_id = add_component(input_panel, component)["id]"

Reply:
similar thing for output_id as well